arch/arm/core/cortex_m/tz/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+# '-mcmse' enables the generation of code for the Secure state of the ARMv8-M
+# Security Extensions. This option is required when building a Secure firmware.
+zephyr_compile_options(-mcmse)
+# '--section-start,.gnu.sgstubs' enables entry veneers (for Secure Entry
+# functions) to be placed inside the special output section .gnu.sgstubs.
+zephyr_ld_option_ifdef(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS
+	${LINKERFLAGPREFIX},--section-start,.gnu.sgstubs=${CONFIG_ARM_NSC_REGION_BASE_ADDRESS}
```

Comment:
"All SG veneers are placed in the special output section .gnu.sgstubs. Its start address must be set, either with the command line option ‘--section-start’ or in a linker script, to indicate where to place these veneers in memory."

In general, it is better to do things in the linker script than in the build system.

If it was attempted and then deemed inappropriate to modify the linker script then documentation
for why we are deviating from this best practice should be documented here.

Reply:
@SebastianBoe, I followed another set of "best practices", been suggested by you, actually :) that is:
- not to pollute core ARM files (such as the arm linker.ld),
- generate as few new files as possible

So, with this approach we satisfy both principles, that is, we don't modify the core ARM linker script, and we don't generate a new file (e.g. a linker-script extension for TrustZone veneers). CMakeLists.txt is, anyway, necessary for all other build options, defined here.
This approach is also enabling us gather all build-time configuration for TrustZone in a single cmake file, which seems good to me.

Reply:
There may be constraints on the region that a user may not know unless they are very familiar with the data sheets.  I am a little unsure about allowing a user to just specify some specific address.  And the default is definitely not right.